src/AutoRest.CSharp/Output/Models/Types/OutputLibrary.cs

```diff
@@ -166,21 +183,48 @@ public OutputLibrary(CodeModel codeModel, BuildContext context)
             return _resourceContainers;
         }
 
+
         public TypeProvider FindTypeForSchema(Schema schema)
         {
-            return SchemaMap[schema];
+            TypeProvider? result;
+            if (!SchemaMap.TryGetValue(schema, out result) && !ResourceSchemaMap.TryGetValue(schema, out result))
+            {
+                throw new KeyNotFoundException($"{schema.Name} was not found in model or resource schema map");
+            }
+            return result;
         }
 
-        private Dictionary<Schema, TypeProvider> SchemaMap => _models ??= BuildModels();
 
         private Dictionary<Schema, TypeProvider> BuildModels()
         {
-            var allSchemas = _codeModel.Schemas.Choices.Cast<Schema>()
-                .Concat(_codeModel.Schemas.SealedChoices)
-                .Concat(_codeModel.Schemas.Objects)
-                .Concat(_codeModel.Schemas.Groups);
+            var models = new Dictionary<Schema, TypeProvider>();
+
+            foreach (var schema in _allSchemas)
+            {
+                /*if (_context.Configuration.AzureArm && _operationGroups.ContainsKey(schema.Name))
+                {
+                    continue;
+                }*/
+                models.Add(schema, BuildModel(schema));
```

Comment:
Reason for this commented out code being here?

Reply:
it will be uncommented in a later PR

Reply:
Because we haven't implemented the codegen for [reosurce]Data, if we skip generating operation resource files here, the generated source code would not be correct.